### PR TITLE
Invalidate collections on question moderation update

### PR DESCRIPTION
**Pull request by npfitz**
Fixes [Question moderation status not updating in collection view](https://www.notion.so/metabase/Verification-on-questions-and-models-not-reflected-in-collections-08f4e2b6b2d24b44bab3a54bd4a07204)

chrome_7AsXaHJfSz

### CODE REVIEW

File: frontend/src/metabase/entities/questions.js [hunk 1]
```diff
@@ -77,6 +80,17 @@ const Questions = createEntity({
   },
 
   reducer: (state = {}, { type, payload, error }) => {
+    if (type === SOFT_RELOAD_CARD) {
+      const { id } = payload;
+      const verified =
+        payload.moderation_reviews?.find(x => x.most_recent)?.status ===
+        "verified"
+          ? "verified"
+          : null;
```
Comment (kulyk) [inline frontend/src/metabase/entities/questions.js:89]: Do we actually need a ternary here? Maybe we should just find the most recent status and add it to the question state if it's not `undefined`? AFAIR `moderated_status` can be something else apart from `verified` and `null`

Comment (kulyk) [reply frontend/src/metabase/entities/questions.js:89]: Should we maybe extract this lookup to a function or break it down a little? I feel it can be a bit hard to follow. Maybe something like:

```js
const latestReview = payload.moderation_reviews?.find(x => x.most_recent)
if (state[id] && latestReview) {
  state[id].moderated_status = latestReview.status
}
```

**Review by kulyk — COMMENTED** [review]
The approach is totally right, given how our entities are synced. I think if we're not missing other statuses except for `"verified"` and could make the code a bit more pretty

File: frontend/src/metabase/entities/questions.js (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/entities/questions.js:89]: @npfitz you would also need to return a new state object without modifying the existing one. Maybe something like this: https://github.com/metabase/metabase/blob/501f4d1a50dbe03d64cdfeee86a87d680eab3d33/frontend/src/metabase/entities/bookmarks.js#L66

**Review by ranquild — APPROVED** [review]
Approving the PR since the approach looks good. Please take a look at the comments

File: frontend/src/metabase/entities/questions.js (hunk 1 above)
Comment (npfitz) [reply frontend/src/metabase/entities/questions.js:89]: Awesome! Will do

File: frontend/src/metabase/entities/questions.js [hunk 2]
```diff
@@ -77,6 +81,17 @@ const Questions = createEntity({
...
   reducer: (state = {}, { type, payload, error }) => {
+    if (type === SOFT_RELOAD_CARD) {
+      const { id } = payload;
+      const latestReview = payload.moderation_reviews?.find(x => x.most_recent);
+
+      if (latestReview) {
+        return updateIn(state, [id], q => ({
+          ...q,
+          moderated_status: latestReview.status,
+        }));
```
Comment (kulyk) [inline frontend/src/metabase/entities/questions.js:92]: Let's maybe use `question` instead of just `q` here? It might not be immediately obvious what it means

**Review by kulyk — APPROVED** [review]
LGTM, just a minor naming thing
